htlcswitch/switch.go

```diff
@@ -3041,3 +2949,162 @@ func (s *Switch) handlePacketAdd(packet *htlcPacket,
 
 	return destination.handleSwitchPacket(packet)
 }
+
+// handlePacketAdd handles forwarding a settle packet.
+func (s *Switch) handlePacketSettle(packet *htlcPacket) error {
+	// If the source of this packet has not been set, use the circuit map
+	// to lookup the origin.
```

Comment:
Q: when can this happen, would be good to describe it in the comment.

Reply:
Cool let me dig a bit...